docs/make.jl

```diff
@@ -14,32 +19,59 @@ function link_example(content)
     return content
 end
 
-for file in readdir(_EXAMPLE_DIR)
-    if !endswith(file, ".jl")
-        continue
-    end
-    filename = joinpath(_EXAMPLE_DIR, file)
-    # `include` the file to test it before `#src` lines are removed. It is in a
-    # testset to isolate local variables between files.
-    @testset "$(file)" begin
-        include(filename)
+function literate_examples()
+    for file in readdir(_EXAMPLE_DIR)
+        if !endswith(file, ".jl")
+            continue
+        end
+        filename = joinpath(_EXAMPLE_DIR, file)
+        # `include` the file to test it before `#src` lines are removed. It is
+        # in a testset to isolate local variables between files.
+        @testset "$(file)" begin
+            include(filename)
+        end
+        Literate.markdown(
+            filename,
+            _EXAMPLE_DIR;
+            documenter = true,
+            postprocess = link_example,
+        )
     end
-    Literate.markdown(
-        filename, _EXAMPLE_DIR; documenter = true, postprocess = link_example
-    )
+    return nothing
+end
+
+if !_FAST
+    literate_examples()
 end
 
 makedocs(
     sitename = "JuMP",
+    authors = "Miles Lubin, Iain Dunning, and Joey Huchette",
     format = Documenter.HTML(
         # See https://github.com/JuliaDocs/Documenter.jl/issues/868
         prettyurls = get(ENV, "CI", nothing) == "true",
         analytics = "UA-44252521-1",
         collapselevel = 1,
     ),
-    # See https://github.com/jump-dev/JuMP.jl/issues/1576
+    # `strict = true` causes Documenter to throw an error if the Doctests fail.
     strict = true,
-    authors = "Miles Lubin, Iain Dunning, and Joey Huchette",
+    # ==========================================================================
+    # `modules = [JuMP]`, along with `checkdocs = :exports` causes Documenter to
+    # thow an error if exported functions with docstrings are not contained in
+    # the Documentation. However, problematically, we include some MOI docs,
+    # which forces us to include MOI in `modules`, despite the fact that we
+    # don't necessarily want to document every MOI method.
+    #
+    # This is should be fine for now, because MOI doesn't export anything.
+    # However, also problematically, some doctests in MOI are not checked and
+    # are failing. Until they are fixed, we can't enable these options.
+    #
+    # TODO(odow): uncomment when possible.
+    # modules = [JuMP, MOI],
+    # checkdocs = :exports,
+    # ==========================================================================
```

Comment:
So we really want to enable these two options. But there are some failures in MOI doctests that I need to fix first.